modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/mapping/smallcluster/SmallClusterFactory.java

```diff
@@ -64,17 +68,24 @@ public ExecutionTarget someOf(List<String> nodes) {
     }
 
     @Override
-    public ExecutionTarget partitioned(List<NodeWithConsistencyToken> nodes) {
-        long[] partitionNodes = new long[nodes.size()];
-        long[] enlistmentConsistencyTokens = new long[nodes.size()];
+    public ExecutionTarget partitioned(List<TokenizedAssignments> assignments) {
+        long[] partitionNodes = new long[assignments.size()];
+        long[] enlistmentConsistencyTokens = new long[assignments.size()];
 
         int idx = 0;
-        for (NodeWithConsistencyToken e : nodes) {
-            partitionNodes[idx] = nodeNameToId.getOrDefault(e.name(), 0);
-            enlistmentConsistencyTokens[idx++] = e.enlistmentConsistencyToken();
+        boolean finalised = true;
+        for (TokenizedAssignments assignment : assignments) {
+            finalised = finalised && assignment.nodes().size() < 2;
+
+            for (Assignment a : assignment.nodes()) {
+                partitionNodes[idx] |= nodeNameToId.getOrDefault(a.consistentId(), 0);
```

Comment:
Default value of zero make no sense.
```suggestion
                long node = nodeNameToId.getOrDefault(a.consistentId(), -1);
                assert partitionNodes[idx] >= 0 : "invalid node";
                partitionNodes[idx] |= node;
```